src/PostgREST/Config.hs

```diff
@@ -334,14 +333,12 @@ readAppConfig :: [(Text, Text)] -> Environment -> Maybe FilePath -> IO (Either T
 readAppConfig dbSettings env optPath = do
   -- Now read the actual config file
   conf <- case optPath of
-    Just cfgPath -> C.load cfgPath `catches`
-      [ Handler (\(ex :: IOError)    -> panic $ "Cannot open config file: " <> show ex)
-      , Handler (\(C.ParseError err) -> panic $ "Error parsing config file: " <> err)
-      ]
+    -- Both C.ParseError and IOError are shown here
+    Just cfgPath -> mapLeft (("Bad config file. " <>) . show) <$> (try $ C.load cfgPath :: IO (Either SomeException C.Config))
```

Comment:
Hm. It looks like we currently don't have a test that checks the output of that. Is the error message still different somehow for `ParseError` and `IOError` - i.e. is it possible to tell from the error message what the problem is? (not sure what `show` will tell us here)

Reply:
Yeah, I already tested that manually. The errors are different:

```
-- For ParseError
postgrest: FatalError {fatalErrorMessage = "Bad config file. ParseError \"support/configs/config:1:14:\\n  |\\n1 | qwerqwerqwer qwer qwer ewqr qew rewq q
ewr\\n  |              ^\\nunexpected 'q'\\nexpecting '=' or '{'\\n\""}

-- For IOError. unexistent file
postgrest: FatalError {fatalErrorMessage = "Bad config file. Cannot open config file: support/configs/configx: openFile: does not exist (No such file or directory)"}
```

Reply:
Ah cool. You know what: I think we don't even need the `Bad config file.` prefix. Those two error messages are good like they are? Almost at least, it would be good to have a nicer `show` instance for `ParseError` without all the `\"` and `\n`?

Reply:
How would I test the error message?

I tried with the `cli` function but I get a PostgrestError exception.

Also, seems `fixtures.yaml` already test this somehow? I see:
```
  - name: non-existant config file
    expect: error
    args: ['does_not_exist.conf']
  - name: invalid config file
```

Reply:
Actually `FatalError` is probably the one that needs better `show`?

Reply:
> How would I test the error message?
> 
> I tried with the `cli` function but I get a PostgrestError exception.
> 
> Also, seems `fixtures.yaml` already test this somehow? I see:
> 
> ```
>   - name: non-existant config file
>     expect: error
>     args: ['does_not_exist.conf']
>   - name: invalid config file
> ```

Yes, this was mainly for coverage. I didn't care too much about the actual output in those cases, yet. This is something that is not implemented, yet. I suggest to leave that for later, because we should add a bit of test infrastructure here and test other error messages, too.

Reply:
>   I suggest to leave that for later, because we should add a bit of test infrastructure here and test other error messages, too.

Cool. Then I'll just remove the `Bad config file` prefix for now.

Reply:
I left a note in #1719, so we can remember to improve the test-cases here.